### PR TITLE
edit cards

### PR DESCRIPTION
**Pull request by danimontanaro**

### CODE REVIEW

File: pages/api/cards.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      update: {
+        answer: newCard.answer,
+        question: newCard.question,
+      },
+      create: {
+        question: newCard.question,
+        answer: newCard.answer,
+        nextReviewDate: newCard.selectedCard.nextReviewDate,
+        deckId: newCard.selectedCard.deckId,
+        level: newCard.selectedCard.level,
```
Comment (davidpaley) [inline pages/api/cards.ts]: maybe you could deconstruct newCard here

File: pages/edit/decks/[deckId].tsx [hunk 2]
```diff
@@ -1,102 +1,133 @@
...
+      onError: (err) => {
+        console.log(err);
+      },
+      // Always refetch after error or success:
+      onSettled: () => {
+        queryClient.invalidateQueries([CARD_QUERY]);
+      },
+
+      onSuccess: () => {
+        //ver que hago si es success
```
Comment (davidpaley) [inline pages/edit/decks/[deckId].tsx]: We should select the new card that was created/updated

File: src/editDeck/deleteCards.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        console.log(err);
+      },
+      // Always refetch after error or success:
+      onSettled: () => {
+        queryClient.invalidateQueries([CARD_QUERY]);
+      },
+
+      onSuccess: () => {
+        setSelectedCardWhenDeleteCard();
+        //ver que hago si es success
```
Comment (davidpaley) [inline src/editDeck/deleteCards.tsx]: delete comment

File: src/editDeck/deleteCards.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <>
+      <Button
+        danger
+        className={styles.deleteCardButton}
+        icon={<DeleteOutlined />}
+        onClick={handleDeleteCardSelected}
+      >
+        Delete Card
+      </Button>
+    </>
```
Comment (davidpaley) [inline src/editDeck/deleteCards.tsx:64]: This should be a modal. And maybe you could change the name to be DeleteCardModal

File: src/editDeck/formCards.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  handleFormSubmit: (value: FormParams) => void;
+}
+
+const FormCardsEdit = ({
+  handleFormSubmit,
+  selectedCard,
+  form,
+}: FormCardEditArgs) => {
+  return (
+    <div>
```
Comment (davidpaley) [inline src/editDeck/formCards.tsx]: why do you need this div?

File: src/editDeck/siderCards.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      question: "  ",
+      answer: "  ",
+      selectedCard: {
+        nextReviewDate: new Date(),
+        deckId: selectedCard.deckId,
+        level: 0,
+      },
+    };
+    handleCreateOrUpdateCard(newCard);
+    handleNewCard(newCard);
```
Comment (davidpaley) [inline src/editDeck/siderCards.tsx]: You shouldn't update the state here but on the success function of React Query

**Review by davidpaley — CHANGES_REQUESTED** [review]

File: pages/api/cards.ts (hunk 1 above)
Comment (danimontanaro) [reply pages/api/cards.ts]: Done!

File: src/editDeck/deleteCards.tsx (hunk 4 above)
Comment (danimontanaro) [reply src/editDeck/deleteCards.tsx:64]: Done! Thanks

File: pages/edit/decks/[deckId].tsx [hunk 7]
```diff
@@ -1,102 +1,128 @@
...
+          handleCreateOrUpdateCard={handleCreateOrUpdateCard}
+        />
         <Content>
           <Row align="middle" justify="space-between">
             <Breadcrumb className={styles.breadcrumb}>
-              <Breadcrumb.Item>Name of the Class</Breadcrumb.Item>
-              <Breadcrumb.Item>Name of the Deck</Breadcrumb.Item>
-              <Breadcrumb.Item>Card1</Breadcrumb.Item>
+              <Breadcrumb.Item>{deckForCards.name}</Breadcrumb.Item>
+              <Breadcrumb.Item>{deckForCards.name}</Breadcrumb.Item>
```
Comment (davidpaley) [inline pages/edit/decks/[deckId].tsx:113]: You are setting the name of the deck twice

File: prisma/schema.prisma [hunk 8]
```diff
@@ -43,7 +43,7 @@ model Deck {
 }
 
 model Card {
-  id String @id @default(cuid())
+  id String @id @default(cuid()) @unique
```
Comment (davidpaley) [inline prisma/schema.prisma:46]: I think that we don't need this unique

**Review by davidpaley — APPROVED** [review]
We can update the types in another PR

**Pull request merged**